File: Max Sum Subarray of size K - GFG/max-sum-subarray-of-size-k.cpp
//{ Driver Code Starts
#include<bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends
class Solution{   
public:
    long maximumSumSubarray(int k, vector<int> &nums , int n){
        long maxSum = 0;
        long windowSum = 0;
        for(int i=0; i<k; i++){
            windowSum += nums[i];
        }
        maxSum = windowSum;
        for(int i=k; i<n; i++){
            windowSum = windowSum + nums[i] - nums[i-k];
            maxSum = max(maxSum, windowSum);
        }
        return maxSum;
    }
};

//{ Driver Code Starts.
int main() 
{ 
    int t;
    cin>>t;
    while(t--)
    {
        int N,K;
        cin >> N >> K;;
        vector<int>Arr;
        for(int i=0;i<N;++i){
            int x;
            cin>>x;
            Arr.push_back(x);
        }
        Solution ob;
        cout << ob.maximumSumSubarray(K,Arr,N) << endl;
    }
    return 0; 
} 
// } Driver Code Ends

File: Max Sum Subarray of size K - GFG/max-sum-subarray-of-size-k.java
//{ Driver Code Starts
//Initial Template for Java
import java.io.*;
import java.util.*;

class GFG
{
    public static void main(String args[])throws IOException
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t-- > 0)
        {
            
            int N=sc.nextInt();
            int K=sc.nextInt();
            ArrayList<Integer> Arr = new ArrayList<Integer>(N); 
  
            // Appending new elements at 
            // the end of the list 
            for (int i = 0; i < N; i++) {
                int x = sc.nextInt();
                Arr.add(x); 
            }

            Solution ob = new Solution();
            System.out.println(ob.maximumSumSubarray(K,Arr,N));
        }
    }
}

// } Driver Code Ends


//User function Template for Java
class Solution{
    static long maximumSumSubarray(int k, ArrayList<Integer> nums, int n){
        long windowSum = 0;
        long maxSum = 0;
        for(int i=0; i<k; i++){
            windowSum += nums.get(i);
        }
        maxSum = windowSum;
        for(int i=k; i<n; i++){
            windowSum = windowSum + nums.get(i) - nums.get(i-k);
            maxSum = Math.max(maxSum, windowSum);
        }
        return maxSum;
    }
}